src/cryptonote_core/cryptonote_core.h

```diff
@@ -1140,6 +1152,10 @@ namespace cryptonote
      uint32_t m_sn_public_ip;
      uint16_t m_storage_port;
 
+     /// Time point at which the storage server last pinged us
+     mutable epee::critical_section m_ping_lock;
+     std::chrono::system_clock::time_point m_last_storage_server_ping;
```

Comment:
Might be sensible to set this to launch time/pass the first threshold such that the initial uptime-proof can be sent on start-up, so there's no timing issues on that initial ping. 

Since the launcher is in charge of the processes, you could imagine the initialisation time of the daemon being slower than the storage server. Storage server sends ping and it's lost by the daemon as it is initialising. Or the daemon crashes and automatically relaunches and loses ping state from the storage server etc etc timing issues.

Reply:
Agree, was thinking the same.

Reply:
Changed to update the ping on startup.